tests/tests.py

```diff
@@ -2318,6 +2318,51 @@ def test_40_passive(self):
 
         self.assertEqual(web_count, count)
 
+        for rtype in ['A', 'AAAA']:
+            rec = next(iter(ivre.db.db.passive.get(
+                ivre.db.db.passive.searchdns(dnstype=rtype),
+                sort=[('addr', 1)]
+            )))
+            req = Request('http://%s:%d/cgi/passivedns/%s' % (
+                HTTPD_HOSTNAME,
+                HTTPD_PORT,
+                rec['addr']
+            ))
+            req.add_header('Referer', 'http://%s:%d/' % (HTTPD_HOSTNAME,
+                                                         HTTPD_PORT))
+            udesc = urlopen(req)
+            self.assertEqual(udesc.getcode(), 200)
+            found = False
+            for line in udesc:
+                cur_rec = json.loads(line)
+                self.assertEqual(rec['addr'], cur_rec['addr'])
+                if cur_rec['rtype'] == rtype and \
+                        cur_rec['rrname'] == rec['value']:
+                    found = True
```

Comment:
Once you have understood why some records may lack and "addr" field, I'd suggest that you break once the expected record has been found:
```suggestion
                    found = True
                    break
```

Reply:
Thanks for the suggestion.